catalog/app/containers/Bucket/PackageTree.js

```diff
@@ -499,19 +529,47 @@ export default function PackageTree({
     ).concat(path.endsWith('/') ? Crumb.Sep(<>&nbsp;/</>) : [])
   }, [bucket, name, revision, path, urls])
 
+  const [revisionKey, setRevisionKey] = React.useState(1)
+  const [revisionListKey, setRevisionListKey] = React.useState(1)
+
+  const onRevisionPush = React.useCallback(
+    (res) => {
+      if (res && res.pushed && res.pushed.name === name) {
+        // refresh revision list if a new revision of the current package has been pushed
+        setRevisionListKey(R.inc)
+        if (revision === 'latest') {
+          // when browsing 'latest' revision, also refresh the pacakge view
+          setRevisionKey(R.inc)
+        }
+      }
+    },
+    [name, revision, setRevisionKey, setRevisionListKey],
+  )
+
   return (
     <FileView.Root>
-      {!!bucketCfg && <ExposeLinkedData {...{ bucketCfg, bucket, name, revision }} />}
+      {!!bucketCfg && (
+        <ExposeLinkedData
+          {...{ bucketCfg, bucket, name, revision }}
+          key={`links:${revisionKey}`}
```

Comment:
As I understand correctly, `revision` change is not enough to refresh this component, and you need to explicitly change React `key` for that

Reply:
if revision is "latest", it wont change after the push, so we need to explicitly "cache-bust" things